concent_api/api-integration-additional-verification-test.py

```diff
@@ -0,0 +1,359 @@
+            cluster_consts,
+            reason=message.tasks.SubtaskResultsRejected.REASON.VerificationNegative,
+            report_computed_task_size=file_size,
+            report_computed_task_package_hash=file_check_sum,
+            task_to_compute_size=file_size,
+            task_to_compute_package_hash=file_check_sum,
+        ),
+        headers = {
+            'Content-Type': 'application/octet-stream',
+        },
+        expected_status=200,
+        expected_message_type=message.concents.ServiceRefused.TYPE,
+        expected_content_type='application/octet-stream',
+    )
+
+
+@count_fails
+def test_case_4_test_for_duplicated_request(cluster_consts, cluster_url, test_id):
+    current_time = get_current_utc_timestamp()
+    (subtask_id, task_id) = get_task_id_and_subtask_id(test_id, 'existing_file')
+
+    file_content_1 = task_id
+    file_content_2 = subtask_id
+    file_size_1 = len(file_content_1)
+    file_size_2 = len(file_content_2)
+    file_check_sum_1 = 'sha1:' + hashlib.sha1(file_content_1.encode()).hexdigest()
+    file_check_sum_2 = 'sha1:' + hashlib.sha1(file_content_2.encode()).hexdigest()
+
+    api_request(
+        cluster_url,
+        'send',
+        PROVIDER_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+        get_subtask_results_verify(
+            task_id,
+            subtask_id,
+            current_time,
+            cluster_consts,
+            reason=message.tasks.SubtaskResultsRejected.REASON.VerificationNegative,
+            report_computed_task_size=file_size_1,
+            report_computed_task_package_hash=file_check_sum_1,
+            task_to_compute_size=file_size_2,
+            task_to_compute_package_hash=file_check_sum_2,
+        ),
+        headers = {
+            'Content-Type': 'application/octet-stream',
+        },
+        expected_status=200,
+        expected_message_type=message.concents.AckSubtaskResultsVerify.TYPE,
+        expected_content_type='application/octet-stream',
+    )
+
+    api_request(
+        cluster_url,
+        'send',
+        PROVIDER_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+        get_subtask_results_verify(
+            task_id,
+            subtask_id,
+            current_time,
+            cluster_consts,
+            reason=message.tasks.SubtaskResultsRejected.REASON.VerificationNegative,
+            report_computed_task_size=file_size_1,
+            report_computed_task_package_hash=file_check_sum_1,
+            task_to_compute_size=file_size_2,
+            task_to_compute_package_hash=file_check_sum_2,
+        ),
+        headers = {
+            'Content-Type': 'application/octet-stream',
+        },
+        expected_status=200,
+        expected_message_type=message.concents.ServiceRefused.TYPE,
+        expected_content_type='application/octet-stream',
+    )
+
+
+@count_fails
+def test_case_5_test_requestor_status_postive(cluster_consts, cluster_url, test_id):
```

Comment:
If it is `positive`, why we expect to get `ServiceRefused`? ;) Please change the name.

Reply:
Done